src/types.js

```diff
@@ -100,7 +100,8 @@ export type ApiRequestOptions = {
 };
 
 export type RequestOptions = {
-  transportMode?: ("api" | "cli")[]
+  transportMode?: ("api" | "cli")[],
+  headers?: { [string]: string }
```

Comment:
do we want to introduce HTTP headers into a system that can request data from an HTTP API or a non-HTTP cli? I would think we'd want to hard-code the the API version per-endpoint (confirmed with @timc1 that the intention is to use this to set the API version) so that we can return proper flow-types and all that rather than allowing SDK consumers to specify API versions.

cc @amccloud @bitpshr

Reply:
> I would think we'd want to hard-code the the API version per-endpoint

Agreed. Although it does raise the question of the best way to achieve that when the old version is still live behind a feature flag. I could see a more specific param for that request making sense, eg `includeTextType` which internally is mapped to an API version?

Reply:
I agree with @rickharris that we should continue abstracting the idea of underlying API versions from consumers. I think we should either use a specific API option for the case where old and new APIs are concurrently live for the same endpoint, or just hardcode the needed API version like we do elsewere.

Reply:
@rickharris I agree about the transport agnostic options for the client. We wouldn't want to hardcode though, as we'll still want to experiment without forcing the new version as the default

@bitpshr I like that second option. Version number is how we reason about these changes elsewhere so seems ideal to keep it the same. @tommoor would this work?

```suggestion
  version?:  number
```

Reply:
It would, but I'd still prefer to keep version number concerns internal to the SDK and instead expose it as a named option relevant to the feature.

Whatever crowd consensus is though.

Reply:
@tommoor that sounds nice but ultimately might not be how we want to end up managing experimental features in the sdk. If you had a bunch of flags that can be combined dynamically in adhoc ways, it would be hard to provide the correct type for that endpoint

My vote would be to add a `version` property (or `_version`) but not document it as public API. The public interface features is controlled by the sdk's semver. We could introduce things to the sdk as the become ready, typed, documented, and out of the box without the need to set flags, verisons, etc.